File: vehicle-tracker-backend/vehicle-service/src/main/java/com/github/diogocerqueiralima/presentation/http/dto/CreateDeviceDTO.java
package com.github.diogocerqueiralima.presentation.http.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

public record CreateDeviceDTO(
        String imei, @JsonProperty("serial_number") String serialNumber, String manufacturer, UUID vehicleId
) {}

File: vehicle-tracker-mobile/app/src/main/java/com/github/diogocerqueiralima/infrastructure/dto/CreateDeviceDTO.kt
package com.github.diogocerqueiralima.infrastructure.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Data class representing the payload sent to the Asset Service to create a device.
 */
@Serializable
data class CreateDeviceDTO(

    @SerialName("serial_number") val serialNumber: String,
    val model: String,
    val manufacturer: String,
    val imei: String,
    @SerialName("owner_id") val ownerId: String

)
